packages/flutter/test/material/dropdown_menu_form_field_test.dart

```diff
@@ -967,6 +968,83 @@ void main() {
     expect(fieldKey.currentState!.value, MenuItem.menuItem0);
   });
 
+  // Regression test for https://github.com/flutter/flutter/issues/174578.
+  testWidgets(
+    'Inner text field is cleared on reset when initialSelection is null - Default controller',
+    (WidgetTester tester) async {
+      final GlobalKey<FormFieldState<MenuItem>> fieldKey = GlobalKey<FormFieldState<MenuItem>>();
+
+      await tester.pumpWidget(
+        MaterialApp(
+          home: Scaffold(
+            body: DropdownMenuFormField<MenuItem>(key: fieldKey, dropdownMenuEntries: menuEntries),
+          ),
+        ),
+      );
+
+      // Select menuItem1.
+      await tester.tap(find.byType(DropdownMenu<MenuItem>));
+      await tester.pump();
+      await tester.tap(findMenuItem(MenuItem.menuItem1));
+      await tester.pump();
+      expect(fieldKey.currentState!.value, MenuItem.menuItem1);
+
+      // After reset the text field content is cleared.
+      fieldKey.currentState!.reset();
+      await tester.pump();
+
+      expect(fieldKey.currentState!.value, null);
+      final TextField textField = tester.widget(find.byType(TextField));
+      expect(
+        textField.controller?.value,
+        const TextEditingValue(selection: TextSelection.collapsed(offset: 0)),
+      );
```

Comment:
Nit: Do you think it would add value to `expect` that the value is not empty before the call to `reset`?

Reply:
Same comment for the next test.